src/dragonball/src/metric.rs

```diff
@@ -2,15 +2,16 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
 use dbs_utils::metric::SharedIncMetric;
 use lazy_static::lazy_static;
 use serde::Serialize;
 
-pub use dbs_utils::metric::IncMetric;
-
 lazy_static! {
     /// Static instance used for handling metrics.
-    pub static ref METRICS: DragonballMetrics = DragonballMetrics::default();
+    pub static ref METRICS: RwLock<DragonballMetrics> = RwLock::new(DragonballMetrics::default());
```

Comment:
Why do you introduce a `RwLock` here?
Is that because you want every vcpu to have its own VcpuMetrics?

Reply:
Yes but not only for vcpu.There will be at least eleven device metric types in DragonballMetrics, such as legacy devices, virtio devices, upcall, etc.